bluemira/codes/_freecadapi.py

```diff
@@ -667,6 +667,33 @@ def bounding_box(obj: apiShape) -> Tuple[float, float, float, float, float, floa
     return box.XMin, box.YMin, box.ZMin, box.XMax, box.YMax, box.ZMax
 
 
+def tessellate(obj: apiShape, tolerance: float) -> None:
+    """
+    Tessellate a geometry object.
+
+    Parameters
+    ----------
+    obj:
+        Shape to tessellate
+    tolerance:
+        Tolerance with which to perform the operation
+
+    Raises
+    ------
+    ValueError:
+        If the tolerance is <= 0.0
+
+    Notes
+    -----
+    Once tesselated an object's properties may change. Tesselation cannot be reverted
+    to a previous lower value, but can be increased (irreversibly).
```

Comment:
We currently use tessellate in the `collect_verts_faces` function for polyscope viewing. It does not copy the shape currently as it was not functionality exposed easily to the user. With this its now possible to get some weird ordering conditions with different tolerances. 

I think some more things need to be done if this is the way forward and I'm tentative about just adding a copy into polyscope because of speed implications.

Its a weird issue that it doesnt every warn if you try to decrease the tesselation otherwise this would be fine.

Reply:
OK, can confirm that appearance is irreversibly modified if tessellate is called with a higher tolerance than the default used in PolyScope. Visually this does not appear to be a problem, but when you turn off the smoothing it's immediately apparent.

Red shape: tessellation = 0.05
Blue shape: tessellation = 0.001

![image](https://user-images.githubusercontent.com/26097289/237060435-92db0f15-1e0e-4ad1-a8ae-05d07fb62276.png)

```python
from bluemira.geometry.tools import *
from bluemira.geometry.face import BluemiraFace
from bluemira.geometry.parameterisations import PolySpline
from bluemira.display import show_cad
from bluemira.display.displayer import DisplayCADOptions
from bluemira.base.components import PhysicalComponent, Component

w = PolySpline().create_shape()
f = BluemiraFace(w)
s = revolve_shape(f, degree=180)  # default tessellation = 0.05
s2 = s.deepcopy()
s2.rotate(degree=180)
s2._tessellate(tolerance=0.001)
s2._tessellate(tolerance=0.05)

from bluemira.display.displayer import ViewerBackend

p1 = PhysicalComponent("t005", s)
p1.display_cad_options = DisplayCADOptions(colour="r", backend=ViewerBackend.POLYSCOPE, tesselation=0.05)
p2 = PhysicalComponent("t0001", s2)
p2.display_cad_options = DisplayCADOptions(colour="b")
component = Component("test", children=[p1, p2])
component.show_cad(backend="POLYSCOPE")
```

will investigate getting the bounding box by tessellating a deepcopied version of `BluemiraGeo` such that the properties are not irreversibly modified and appearance is more consistent when viewed in PolyScope